src/stdlib_strings.f90

```diff
@@ -499,5 +513,140 @@ pure function compute_lps(string) result(lps_array)
     
     end function compute_lps
 
+    !> Replaces all occurrences of substring 'pattern' in the input 'string'
+    !> with the replacement 'replacement'
+    !> Returns a new string
+    pure function replace_all_string_string_string(string, pattern, replacement) result(res)
+        type(string_type), intent(in) :: string
+        type(string_type), intent(in) :: pattern
+        type(string_type), intent(in) :: replacement
+        type(string_type) :: res
+
+        res = string_type(replace_all_char_char_char(char(string), & 
```

Comment:
Should I call `replace_all` instead of `replace_all_char_char_char` in line 525?

Reply:
It's preferable to call the generic interface here